libs/barista-components/copy-to-clipboard/src/copy-to-clipboard.ts

```diff
@@ -89,26 +93,25 @@ export class DtCopyToClipboard implements AfterContentInit, OnDestroy {
   /** Copies the provided content to the clipboard. */
   copyToClipboard(): void {
     if (this._input) {
-      this._input.nativeElement.select();
-      const copyResult = document.execCommand('copy');
-      if (!copyResult) {
-        this.copyFailed.emit();
-        return;
-      }
-      this._showIcon = true;
-      _addCssClass(this._input.nativeElement, DT_COPY_TO_CLIPBOARD_SUCCESSFUL);
-      if (this._copyButton) {
-        _addCssClass(
-          this._copyButton.nativeElement,
-          DT_COPY_TO_CLIPBOARD_SUCCESSFUL,
-        );
-        this._copyButton.nativeElement.focus();
-      }
+      this._input.nativeElement.textContent !== null
+        ? this.cdkClipboard.copy(this._input.nativeElement.textContent)
+        : this.copyFailed.emit();
+    }
+  }
+
+  copiedToClipboard(): void {
```

Comment:
does this one need to be public?

Reply:
Its called by the CDK when it finished copying, therefor I'll annotate it with @internal in the new commit.